client/client.go

```diff
@@ -579,6 +799,48 @@ func (c *Client) Call(ctx context.Context, procedure string, options wamp.Dict,
 
 	switch msg := msg.(type) {
 	case *wamp.Result:
+
+		// Check and handle Payload PassThru Mode
+		// @see https://wamp-proto.org/wamp_latest_ietf.html#name-payload-passthru-mode
+		if pptScheme, _ := msg.Details[wamp.OptPPTScheme].(string); pptScheme != "" {
+			// Let's check: was ppt feature announced by dealer?
+			if !c.sess.HasFeature(wamp.RoleDealer, wamp.FeaturePayloadPassthruMode) {
+				// It's protocol violation, so we need to abort connection
+				abortMsg := wamp.Abort{
+					Reason: wamp.ErrProtocolViolation,
+					Details: wamp.Dict{
+						wamp.OptError:   ErrPPTNotSupportedByRouter.Error(),
+						wamp.OptMessage: ErrPPTNotSupportedByPeer.Error(),
+					},
+				}
+				c.sess.Peer.Send(&abortMsg)
+				c.sess.Peer.Close()
+				return nil, fmt.Errorf("cannot process unexpected passthrough result: %v", ErrPPTNotSupportedByRouter)
```

Comment:
It probably makes sense to wrap errors, so that the type of error can be tested at a higher level.
```suggestion
				return nil, fmt.Errorf("cannot process unexpected passthrough result: %w", ErrPPTNotSupportedByRouter)
```

This is something that can likely be done throughout the code, and changes to code outside of this PR can be done separately.

Reply:
yeah! I want to setup golint-ci and fix most warnings.